internal/pkg/api/server/templates/service/service.go

```diff
@@ -314,19 +319,51 @@ func (s *service) UpgradeInstanceValidateInputs(dependencies.Container, *Upgrade
 	return nil, NotImplementedError{}
 }
 
-func repositories() []model.TemplateRepository {
-	return []model.TemplateRepository{
-		{
-			Type: "git",
-			Name: repository.DefaultTemplateRepositoryName,
-			Url:  repository.DefaultTemplateRepositoryUrl,
-			Ref:  "api-demo",
-		},
-	}
+func repositories(repoPath string) ([]model.TemplateRepository, error) {
+	if repoPath == "" {
+		return []model.TemplateRepository{
+			{
+				Type: "git",
+				Name: repository.DefaultTemplateRepositoryName,
+				Url:  repository.DefaultTemplateRepositoryUrl,
+				Ref:  "api-demo",
+			},
+		}, nil
+	}
+
+	if strings.HasPrefix(repoPath, "file://") {
+		return []model.TemplateRepository{
+			{
+				Type: "dir",
+				Name: repository.DefaultTemplateRepositoryName,
+				Path: strings.TrimPrefix(repoPath, "file://"),
+			},
+		}, nil
+	}
+	if strings.HasPrefix(repoPath, "https://") {
+		uri := strings.TrimPrefix(repoPath, "https://")
+		refDividerPos := strings.Index(uri, ":")
+		if refDividerPos < 0 {
+			return nil, fmt.Errorf("invalid repository path url, no ref provided")
+		}
+		return []model.TemplateRepository{
+			{
+				Type: "git",
+				Name: repository.DefaultTemplateRepositoryName,
+				Url:  "https://" + uri[:refDividerPos],
+				Ref:  uri[refDividerPos+1:],
+			},
+		}, nil
+	}
```

Comment:
No nejsem si jistej jestli je tady to správný místo ale jinam to asi dát nemůžu. (?) 🤔